### PR TITLE
Review all of my changes

### PR DESCRIPTION
**Pull request by maxammann**

### CODE REVIEW

**Comment by djc** [comment]
This is way too large to review in one go. If you want to submit modifications to rustls, please submit one PR per logical change with a clearly explained rationale or use case.

File: rustls/src/msgs/handshake.rs [hunk 1]
```diff
@@ -1222,7 +1232,11 @@ impl Codec for ServerHelloPayload {
         let session_id = SessionID::read(r)?;
         let suite = CipherSuite::read(r)?;
         let compression = Compression::read(r)?;
-        let extensions = codec::read_vec_u16::<ServerExtension>(r)?;
+        let extensions = if r.any_left() {
+            codec::read_vec_u16::<ServerExtension>(r)?
+        } else {
+            vec![]
+        };
```
Comment (maxammann) [inline rustls/src/msgs/handshake.rs:1239]: This is werid

File: rustls/src/msgs/handshake.rs [hunk 2]
```diff
@@ -1562,11 +1576,6 @@ impl Codec for ECParameters {
 
     fn read(r: &mut Reader) -> Option<Self> {
         let ct = ECCurveType::read(r)?;
-
-        if ct != ECCurveType::NamedCurve {
-            return None;
-        }
-
```
Comment (maxammann) [inline rustls/src/msgs/handshake.rs:1569]: Probably removal of a check

File: rustls/src/msgs/ccs.rs [hunk 3]
```diff
@@ -11,10 +11,6 @@ impl Codec for ChangeCipherSpecPayload {
     fn read(r: &mut Reader) -> Option<Self> {
         let typ = u8::read(r)?;
 
-        if typ == 1 && !r.any_left() {
-            Some(Self {})
-        } else {
-            None
-        }
```
Comment (maxammann) [inline rustls/src/msgs/ccs.rs:18]: Probably removal of a check

**Comment by maxammann** [comment]
> This is way too large to review in one go. If you want to submit modifications to rustls, please submit one PR per logical change with a clearly explained rationale or use case.

Uff, I'm very sorry, this shouldn't have been created in this repo. Sorry!

**Pull request closed without merging**